src/App.tsx

```diff
@@ -1,24 +1,228 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
+import { Todo } from './types/Todo';
 
-const USER_ID = 0;
+import {
+  addTodo, deleteTodo, getTodos, updateTodo,
+} from './api/todos';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { FilterType } from './types/FilterType';
+import { Footer } from './components/Footer';
+import { Notification } from './components/Notification';
+import { getFilteredTodos } from './utils/getFilteredTodos';
+import { UserWarning } from './UserWarning';
+import { getActiveTodos } from './utils/getActiveTodos';
+import { getCompletedTodos } from './utils/getCompletedTodos';
+import { ErrorType } from './types/ErrorType';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState<FilterType>(FilterType.ALL);
+  const [hasError, setHasError] = useState(false);
+  const [error, setError] = useState<ErrorType>(ErrorType.NO_ERROR);
+  const [isInputDisabled, setIsInputDisabled] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [activeTodosId, setActiveTodosId] = useState<number[]>([]);
+
+  const USER_ID = 6381;
+
+  const getAllTodos = async (userId: number) => {
+    try {
+      const todosFromServer = await getTodos(userId);
+
+      setTodos(todosFromServer);
+    } catch {
+      setHasError(true);
+      setError(ErrorType.UPLOAD_ERROR);
+      // eslint-disable-next-line no-console
+      console.warn('An occur error while loading todos');
+    }
+  };
+
+  useEffect(() => {
+    getAllTodos(USER_ID);
+  }, []);
+
+  const activeTodos = useMemo(() => getActiveTodos(todos), [todos]);
+
+  const completedTodos = useMemo(() => getCompletedTodos(todos), [todos]);
+
+  const handleAddTodo = async (userId: number, title: string) => {
+    if (!title.trim()) {
+      setHasError(true);
+      setError(ErrorType.EMPTY_ERROR);
+
+      return;
+    }
+
+    const newTodo = {
+      id: 0,
```

Comment:
no no no, `0` isn't valid value for id, you shouldn't send id for todo on server